internal/pkg/cli/pipeline_show.go

```diff
@@ -87,34 +87,35 @@ func newShowPipelineOpts(vars showPipelineVars) (*showPipelineOpts, error) {
 	return opts, nil
 }
 
-// Validate returns an error if the flag values passed by the user are invalid.
+// Validate returns an error if the optional flag values passed by the user are invalid.
 func (o *showPipelineOpts) Validate() error {
 	if o.appName != "" {
 		if _, err := o.store.GetApplication(o.appName); err != nil {
-			return err
+			return fmt.Errorf("validate application name: %w", err)
 		}
-	}
-	if o.pipelineName != "" {
-		if _, err := o.pipelineSvc.GetPipeline(o.pipelineName); err != nil {
+	} else {
+		if err := o.askAppName(); err != nil {
```

Comment:
should we move this to `Ask` ? I feel like if we're going to prompt for it that means that the flag is required 💭

Reply:
Yeah, you're right. I was considering it optional bc we `tryReadingAppName()`, but it didn't sit well with me because yeah-- it's not optional in that we do need it to proceed. Moving.

Reply:
Actually, this is part of a larger bug with `pipeline show` and `pipeline status` that I've been wanting to fix for a long time. I'll add changes to this PR. If there is a pipeline manifest in the workspace from which the command is run, we show info/status for that workspace pipeline regardless of `o.appName`. That's because we first read the manifest for the pipeline name/other info, independently of the app name. We only use the app name if there isn't a detected pipeline manifest, at which point we retrieve pipeline names with the app name tag.

Reply:
Just a callout here: it seems like `appName` is only needed when we ask for pipeline name (we use `appName` to list out all the pipelines of the app). It however doesn't play a role in `Execute`. For this reason, I wouldn't consider `appName` "required". 

We should probably ask for `app` only when `o.name == ""` in `Ask`.

Reply:
Okay. I moved validation of passed-in appName back into `Validate()`, and only ask for `appName` if we're also asking for `name`. 

The code no longer reads the manifest for the pipeline name, because `show` and `status` are only concerned about deployed pipelines. Before and after manual output in comment below.